app/controllers/api/subcollections/alert_definitions.rb

```diff
@@ -4,6 +4,15 @@ module AlertDefinitions
       def alert_definitions_query_resource(object)
         object.respond_to?(:miq_alerts) ? object.miq_alerts : []
       end
+
+      def alert_definitions_assign_resource(object, _type, id = nil, _data = nil)
+        object.add_member(MiqAlert.find(id))
```

Comment:
`MiqAlert.find` will raise a `RecordNotFound` error in the case that this id does not match a record in the db. Can you ensure that there is test coverage for this and the unassign action and that you have the desired behavior?

You may also need to use `result_search` here (which delegates to RBAC), since a bare `find` will ignore all such authorization concerns.